### PR TITLE
Store cursors in `ConnInfo`

### LINKED ISSUE #1733: Support `getMore` command
### What should be done?

https://www.mongodb.com/docs/manual/reference/command/getMore/

- [x] Do not consume all documents in `find`: #2933
- [x] Do not consume all documents in `aggregate`
- [ ] Refactor tests (https://github.com/FerretDB/FerretDB/pull/2929/files#r1246365875)

Test script: #2761

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Extracted from #1769.

Closes #1733.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/clientconn/conninfo/conn_info.go [hunk 1]
```diff
@@ -12,28 +12,142 @@
...
+		lastCursorID.Store(rand.Uint32())
+	}
+}
 
 // ConnInfo represents connection info.
 type ConnInfo struct {
 	PeerAddr string
 
 	rw       sync.RWMutex
+	cursors  map[int64]Cursor
```
Comment (rumyantseva) [inline internal/clientconn/conninfo/conn_info.go:57]: Don't you wan't to use `uint32` as the map key?

File: internal/clientconn/conninfo/conn_info.go [hunk 2]
```diff
@@ -12,28 +12,142 @@
...
+func (connInfo *ConnInfo) StoreCursor(iter iterator.Interface[int, *types.Document], filter *types.Document) int64 {
+	connInfo.rw.Lock()
+	defer connInfo.rw.Unlock()
+
+	var id int64
+
+	// use global, sequential, positive, short cursor IDs to make debugging easier
+	for {
+		id = int64(lastCursorID.Add(1))
+		if _, ok := connInfo.cursors[id]; id != 0 && !ok {
```
Comment (rumyantseva) [inline internal/clientconn/conninfo/conn_info.go:128]: This approach looks helpful indeed!

**Review by rumyantseva — APPROVED** [review]
Looks helpful!

**Review by w84thesun — APPROVED** [review]

File: internal/clientconn/conninfo/conn_info.go (hunk 1 above)
Comment (AlekSi) [reply internal/clientconn/conninfo/conn_info.go:57]: It is int64 in the wire protocol. While we could save a few bytes per cursor, that would make more code depend on the ID generation. For example, consider this code:

```
// Cursor returns cursor by ID, or nil.
func (connInfo *ConnInfo) Cursor(id int64) *Cursor {
	connInfo.rw.RLock()
	defer connInfo.rw.RUnlock()

	c, ok := connInfo.cursors[uint32(id)]
	if !ok {
		return nil
	}

	return &c
}
```

It is subtly wrong: there are two int64 values that correspond to the same uint32 value. So we have to check that id > 0 and <= math.MaxUint32. That would split that logic into two functions. Not sure if it is worth it

Comment (rumyantseva) [reply internal/clientconn/conninfo/conn_info.go:57]: ok, makes sense!

**Pull request merged**